lighthouse-core/audits/byte-efficiency/uses-webp-images.js

```diff
@@ -47,7 +46,9 @@ class UsesWebPImages extends ByteEfficiencyAudit {
   static audit_(artifacts) {
     const images = artifacts.OptimizedImages;
 
+    /** @type {Array<LH.Artifacts.OptimizedImageError>} */
     const failedImages = [];
+    /** @type {Array<{url: string, fromProtocol: boolean, isCrossOrigin: boolean, totalBytes: number, wastedBytes: number}>} */
```

Comment:
since this is used in multiple audits can we share it? i'm okay either way.

Reply:
it's almost but not quite the same as we use in other audits, enough to be annoying to redefine. `url` has to be a string and there's the extra `isCrossOrigin` and `fromProtocol` properties. I don't think it's worth defining externally for just this and `uses-optimized-images`, though, since it just becomes a `ByteEfficiencyResult` to the rest of the pipeline

Reply:
I can make the `failedImages` type be inferred, though